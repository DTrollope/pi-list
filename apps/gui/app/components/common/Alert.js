import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const propTypes = {
    icon: PropTypes.string,
    className: PropTypes.string,
    type: PropTypes.string,
    showIcon: PropTypes.bool,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node
    ]).isRequired
};

const defaultProps = {
    icon: '',
    type: 'danger',
    className: '',
    showIcon: false
};

const Alert = (props) => {
    let icon;

    const alertClassName = classNames(
        'lst-alert',
        'col-xs-12',
        props.className,
        {
            'lst-alert--success': props.type === 'success',
            'lst-alert--danger': props.type === 'danger',
            'lst-alert--warning': props.type === 'warning',
            'lst-alert--info': props.type === 'info',
        }
    );

    const showIcon = props.showIcon || props.icon;

    switch (props.type) {
    case 'success':
        icon = 'check_circle';
        break;
    case 'warning':
        icon = 'warning';
        break;
    case 'info':
        icon = 'info';
        break;
    case 'danger':
    default:
        icon = 'error';
        break;
    }

    if (props.icon) {
        icon = props.icon;
    }

    return (
        <div className={alertClassName}>
            {showIcon && (
                <div className="lst-alert__icon">
                    <i className="lst-alert__icon-size lst-icons">{icon}</i>
                </div>
            )}
            <div className="lst-alert__content">
                {props.children}
            </div>
        </div>
    );
};

Alert.propTypes = propTypes;
Alert.defaultProps = defaultProps;

export default Alert;
