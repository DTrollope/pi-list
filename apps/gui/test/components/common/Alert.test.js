import React from 'react';
import { shallow } from 'enzyme';
import Alert from 'components/common/Alert';
import renderer from 'react-test-renderer';

describe('components/common/Alert', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <Alert>
                Hello World!
            </Alert>
        );
    });

    describe('Alert Component', () => {
        it('should include a custom className', () => {
            wrapper.setProps({ className: 'custom-classname' });

            expect(wrapper.find('.lst-alert').hasClass('custom-classname')).toBe(true);
        });

        it('should render the children test', () => {
            expect(wrapper.find('.lst-alert').text()).toBe('Hello World!');
        });
    });

    describe('Alert Snapshots', () => {
        it('should render a default alert (danger) with the correct label', () => {
            const tree = renderer
                .create(<Alert>Hello world</Alert>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a default alert (danger) with the correct label', () => {
            const tree = renderer
                .create(<Alert>Hello world</Alert>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a danger alert with the correct label', () => {
            const tree = renderer
                .create(<Alert type="danger">Hello world</Alert>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a success alert with the correct label', () => {
            const tree = renderer
                .create(<Alert type="success">Hello world</Alert>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a info alert with the correct label', () => {
            const tree = renderer
                .create(<Alert type="info">Hello world</Alert>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a warning alert with the correct label', () => {
            const tree = renderer
                .create(<Alert type="warning">Hello world</Alert>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a danger alert with the correct label and correct icon', () => {
            const tree = renderer
                .create(<Alert type="danger" showIcon>Hello world</Alert>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a success alert with the correct label and correct icon', () => {
            const tree = renderer
                .create(<Alert type="success" showIcon>Hello world</Alert>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a info alert with the correct label and correct icon', () => {
            const tree = renderer
                .create(<Alert type="info" showIcon>Hello world</Alert>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a warning alert with the correct label and correct icon', () => {
            const tree = renderer
                .create(<Alert type="warning" showIcon>Hello world</Alert>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render an alert with custom icon', () => {
            const tree = renderer
                .create(<Alert type="warning" icon="build">Hello world</Alert>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });
    });
});