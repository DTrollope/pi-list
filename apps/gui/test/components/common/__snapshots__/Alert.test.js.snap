// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/Alert Alert Snapshots should render a danger alert with the correct label 1`] = `
<div
  className="lst-alert col-xs-12 lst-alert--danger"
>
  
  <div
    className="lst-alert__content"
  >
    Hello world
  </div>
</div>
`;

exports[`components/common/Alert Alert Snapshots should render a danger alert with the correct label and correct icon 1`] = `
<div
  className="lst-alert col-xs-12 lst-alert--danger"
>
  <div
    className="lst-alert__icon"
  >
    <i
      className="lst-alert__icon-size lst-icons"
    >
      error
    </i>
  </div>
  <div
    className="lst-alert__content"
  >
    Hello world
  </div>
</div>
`;

exports[`components/common/Alert Alert Snapshots should render a default alert (danger) with the correct label 1`] = `
<div
  className="lst-alert col-xs-12 lst-alert--danger"
>
  
  <div
    className="lst-alert__content"
  >
    Hello world
  </div>
</div>
`;

exports[`components/common/Alert Alert Snapshots should render a default alert (danger) with the correct label 2`] = `
<div
  className="lst-alert col-xs-12 lst-alert--danger"
>
  
  <div
    className="lst-alert__content"
  >
    Hello world
  </div>
</div>
`;

exports[`components/common/Alert Alert Snapshots should render a info alert with the correct label 1`] = `
<div
  className="lst-alert col-xs-12 lst-alert--info"
>
  
  <div
    className="lst-alert__content"
  >
    Hello world
  </div>
</div>
`;

exports[`components/common/Alert Alert Snapshots should render a info alert with the correct label and correct icon 1`] = `
<div
  className="lst-alert col-xs-12 lst-alert--info"
>
  <div
    className="lst-alert__icon"
  >
    <i
      className="lst-alert__icon-size lst-icons"
    >
      info
    </i>
  </div>
  <div
    className="lst-alert__content"
  >
    Hello world
  </div>
</div>
`;

exports[`components/common/Alert Alert Snapshots should render a success alert with the correct label 1`] = `
<div
  className="lst-alert col-xs-12 lst-alert--success"
>
  
  <div
    className="lst-alert__content"
  >
    Hello world
  </div>
</div>
`;

exports[`components/common/Alert Alert Snapshots should render a success alert with the correct label and correct icon 1`] = `
<div
  className="lst-alert col-xs-12 lst-alert--success"
>
  <div
    className="lst-alert__icon"
  >
    <i
      className="lst-alert__icon-size lst-icons"
    >
      check_circle
    </i>
  </div>
  <div
    className="lst-alert__content"
  >
    Hello world
  </div>
</div>
`;

exports[`components/common/Alert Alert Snapshots should render a warning alert with the correct label 1`] = `
<div
  className="lst-alert col-xs-12 lst-alert--warning"
>
  
  <div
    className="lst-alert__content"
  >
    Hello world
  </div>
</div>
`;

exports[`components/common/Alert Alert Snapshots should render a warning alert with the correct label and correct icon 1`] = `
<div
  className="lst-alert col-xs-12 lst-alert--warning"
>
  <div
    className="lst-alert__icon"
  >
    <i
      className="lst-alert__icon-size lst-icons"
    >
      warning
    </i>
  </div>
  <div
    className="lst-alert__content"
  >
    Hello world
  </div>
</div>
`;

exports[`components/common/Alert Alert Snapshots should render an alert with custom icon 1`] = `
<div
  className="lst-alert col-xs-12 lst-alert--warning"
>
  <div
    className="lst-alert__icon"
  >
    <i
      className="lst-alert__icon-size lst-icons"
    >
      build
    </i>
  </div>
  <div
    className="lst-alert__content"
  >
    Hello world
  </div>
</div>
`;
